slimta/logging/http.py

```diff
@@ -24,10 +24,11 @@
 
 """
 
-from __future__ import absolute_import
+from __future__ import absolute_import, unicode_literals
```

Comment:
I'm against having varying string typing for aesthetic reasons, plus it's IMHO good to explicit the fact that WSGI dict keys are bytes for py2 and unicode for py3, which is not something obvious, and the only part of slimta requiring such behaviour IIRC.

To make it a bit more readable, we could do something like :

```diff
diff --git a/slimta/logging/http.py b/slimta/logging/http.py
index c695f04..c9e1159 100644
--- a/slimta/logging/http.py
+++ b/slimta/logging/http.py
@@ -32,6 +32,12 @@ from functools import partial
 __all__ = ['HttpLogger']
 
 
+ENV_REQUEST_METHOD = str('REQUEST_METHOD')
+ENV_PATH_INFO = str('PATH_INFO')
+ENV_CONTENT_TYPE = str('CONTENT_TYPE')
+ENV_CONTENT_LENGTH = str('CONTENT_LENGTH')
+
+
 class HttpLogger(object):
     """Provides a limited set of log methods that :mod:`slimta` packages may
     use. This prevents free-form logs from mixing in with standard, machine-
@@ -46,17 +52,17 @@ class HttpLogger(object):
         self.log = partial(logline, log.debug, str('http'))
 
     def _get_method_from_environ(self, environ):
-        return environ[str('REQUEST_METHOD')].upper()
+        return environ[ENV_REQUEST_METHOD].upper()
 
     def _get_path_from_environ(self, environ):
-        return environ.get(str('PATH_INFO'), None)
+        return environ.get(ENV_PATH_INFO, None)
 
     def _get_headers_from_environ(self, environ):
         ret = []
         for key, value in environ.items():
-            if key == str('CONTENT_TYPE'):
+            if key == ENV_CONTENT_TYPE:
                 ret.append((str('Content-Type'), value))
-            elif key == str('CONTENT_LENGTH'):
+            elif key == ENV_CONTENT_LENGTH:
                 ret.append((str('Content-Length'), value))
             elif key.startswith(str('HTTP_')):
                 parts = key.split(str('_'))
```
… I myself don't think it's necessary, but you'll have final word on that as project maintainer anyway, so you tell me, after this discussion, which solution you choose :)